xds/internal/xdsclient/xdsresource/endpoints_resource_type.go

```diff
@@ -0,0 +1,166 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xdsresource
+
+import (
+	"google.golang.org/grpc/internal/pretty"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	// Compile time interface checks.
+	_ Type         = endpointsResourceType{}
+	_ ResourceData = &EndpointsResourceData{}
+
+	// Singleton instantiation of the resource type implementation.
+	endpointsType = endpointsResourceType{}
+)
+
+// endpointsResourceType provides the resource-type specific functionality for a
+// ClusterLoadAssignment (or Endpoints) resource.
+//
+// Implements the Type interface.
+type endpointsResourceType struct {
+}
+
+// V2TypeURL is the xDS type URL of this resource type for v2 transport.
+func (endpointsResourceType) V2TypeURL() string {
+	return "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment"
+
+}
+
+// V3TypeURL is the xDS type URL of this resource type for v3 transport.
+func (endpointsResourceType) V3TypeURL() string {
+	return "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment"
+}
+
+// TypeEnum identifies resources in a transport protocol agnostic way.
+func (endpointsResourceType) TypeEnum() ResourceType {
+	return EndpointsResource
+}
```

Comment:
This is common to all of these types; consider maybe embedding a struct that implements these and has fields that can be set to these values instead.

Reply:
Good idea. Thanks, done.